xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java

```diff
@@ -466,15 +464,21 @@ private void assertEmptyResolutionResult() {
   }
 
   @SuppressWarnings("unchecked")
-  private static Result assertCallSelectResult(
+  private Result assertCallSelectResult(
```

Comment:
Is the return value still needed? If not, we can change it to void. Or it can be changed to return the ClientCall, so that you can get rid of the global `testCall`. Slightly improves code integrity.

Reply:
The return value is used for some assertions following it:
https://github.com/grpc/grpc-java/blob/774b39c42bb209c5dac80d7701d87d1027763ace/xds/src/test/java/io/grpc/xds/XdsNameResolverTest.java#L454-L455

Returning void makes more sense. Assertion method ideally should not return anything. Unfortunately `assertCallSelectResult ` is ugly in the sense that it does an action (the config selection) and an assertion for some of the state changes, but is followed by another assertion `assertServiceConfigForMethodConfig()` for some other state changes in the `result` of the first assertion method's action. The two assertions should be combined. WDYT?

Reply:
Yeah, it is ugly. That's why I was wondering if the returned value is still needed. If you'd like to merge `assertCallSelectResult()` and `assertServiceConfigForMethodConfig()`, that would be great (it should be easy, only one test method needs to be cleaned up).

Reply:
Also, it would be great to rename `assertCallSelectResult()` to `selectConfig()`. But whatever, I am fine with leaving it...